tinygrad/codegen/ast.py

```diff
@@ -55,7 +55,7 @@ def __init__(self, ast:LazyOp, output_buffer=None):
           break
 
     # create the buffer we are returning (as the same type as the input buffers) and add it as the first buffer
-    self.ret = output_buffer if output_buffer else type(self.bufs[0])(output_shape if output_shape else self.info.shape, force_create=True)
+    self.ret = output_buffer or type(self.bufs[0])(output_shape or self.info.shape, force_create=True)
```

Comment:
Please do one thing in this commit. If this introduces a bug, I'll never find it.

Reply:
rebased with changes of https://github.com/geohot/tinygrad/pull/673 and reverted all formatting that could create buggy corner case.